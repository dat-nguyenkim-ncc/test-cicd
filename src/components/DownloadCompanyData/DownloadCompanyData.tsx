import React, { ChangeEvent, useState } from 'react'
import { Flex } from 'theme-ui'
import { FormOption, ViewInterface } from '../../types'
import { Button, Dropdown, Updating } from '../'
import { Heading, Paragraph, Section } from '../primitives'
import strings from '../../strings'

export type StateDownloadData = {
  mapping: string | null
  category: string | null
}

export type DownloadCompanyDataProps = ViewInterface<{
  categoryOptions: FormOption[]
  mappingOptions: FormOption[]
  lastUpdated: string
  loading?: boolean
  onPressDownload(data: StateDownloadData): void
}>

const DownloadCompanyData = ({
  onPressDownload,
  categoryOptions,
  mappingOptions,
  lastUpdated,
  loading,
  sx,
}: DownloadCompanyDataProps) => {
  const { downloadCompanyData: copy } = strings
  const [state, setState] = useState<StateDownloadData>({
    mapping: mappingOptions[0].value ? (mappingOptions[0].value as string) : null,
    category: categoryOptions[0].value ? (categoryOptions[0].value as string) : null,
  })

  const onChangeDropdown = (e: ChangeEvent<HTMLSelectElement>) => {
    setState({ ...state, [e.currentTarget.name]: e.currentTarget.value })
  }

  const onDownloadPress = () => {
    onPressDownload(state)
  }

  return (
    <Section sx={sx}>
      <Heading as="h3">{copy.heading}</Heading>
      <Paragraph sx={{ mt: 6, maxWidth: 380 }}>{copy.downloadData}</Paragraph>
      <Flex mt={6} sx={{ justifyContent: 'space-between' }}>
        <Paragraph bold>{copy.filterBy}</Paragraph>
        <Dropdown
          onChange={onChangeDropdown}
          name="category"
          options={categoryOptions}
          label={copy.dropdowns.category}
          value={state.category as string}
        />
        <Flex
          sx={{
            flex: 0.7,
            width: 274,
            height: 76,
            justifyContent: 'flex-end',
            alignItems: 'center',
          }}
        >
          {!loading ? (
            <Button
              onPress={onDownloadPress}
              size="big"
              label={copy.downloadButton}
              icon="download"
            />
          ) : (
            <Updating noPadding loading />
          )}
        </Flex>
      </Flex>
    </Section>
  )
}

export default DownloadCompanyData
