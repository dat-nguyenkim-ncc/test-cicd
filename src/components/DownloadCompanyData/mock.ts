export const categoryOptions = [
  { value: 'fintech', label: 'Fintech' },
  { value: 'regtech', label: 'Regtech' },
  { value: 'insuretech', label: 'Insuretech' },
  { value: 'all', label: 'All' },
]

export const mappingOptions = [
  { value: 'fintech', label: 'Fintech' },
  { value: 'regtech', label: 'Regtech' },
  { value: 'insuretech', label: 'Insuretech' },
  { value: 'all', label: 'All' },
]
