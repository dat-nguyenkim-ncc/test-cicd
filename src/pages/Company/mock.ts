import {
  companyDetails,
  companyFinancials,
  companyBusiness,
  companyPeople,
  mappingSummary,
  companySources,
  companyOverview,
} from '../../__mock__'

export const mockEditCompany = {
  details: companyDetails(),
  overview: companyOverview(),
  financials: companyFinancials(),
  business: companyBusiness(),
  people: companyPeople(),
  mapping: mappingSummary,
}

export const mockAddCompany = {
  details: companyDetails(),
  overview: companyOverview(),
  financials: companyFinancials(),
}

export const sources = companySources
