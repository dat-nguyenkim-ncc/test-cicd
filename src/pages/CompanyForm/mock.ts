import { FormOption } from '../../types'
import {
  EnumBoolean,
  EnumCompanyNewsStatus,
  EnumCompanyNewsStatusId,
  EnumExpandStatus,
  EnumExpandStatusId,
  EnumInvestor,
} from '../../types/enums'

export const DEFAULT_AVATAR =
  'https://s3.eu-west-2.amazonaws.com/dealroom-static/null/people-32.png'

// TODO get this from backend
export const regions = [
  { label: 'Europe', value: 'europe' },
  { label: 'North America', value: 'na' },
  { label: 'South America', value: 'sa' },
  { label: 'Central America', value: 'ca' },
  { label: 'Asia', value: 'asia' },
  { label: 'Australia', value: 'australia' },
]

export const countries = [
  { label: 'UK', value: 'uk' },
  { label: 'Brazil', value: 'br' },
  { label: 'Peru', value: 'peru' },
  { label: 'Spain', value: 'spain' },
]

export enum EStatusValue {
  ACQUIRED = 'Acquired',
  CLOSED = 'Closed',
  IPO = 'IPO / Went public',
  MERGED = 'Merged',
  OPERATING = 'Operating',
}

export const status = [
  { label: 'Acquired', value: EStatusValue.ACQUIRED },
  { label: 'Closed', value: EStatusValue.CLOSED },
  { label: 'IPO / Went public', value: EStatusValue.IPO },
  { label: 'Merged', value: EStatusValue.MERGED },
  { label: 'Operating', value: EStatusValue.OPERATING },
]

export const expandStatus = [
  { value: EnumExpandStatusId.FOLLOWING, label: EnumExpandStatus.FOLLOWING },
  { value: EnumExpandStatusId.DUPLICATED, label: EnumExpandStatus.DUPLICATED },
  { value: EnumExpandStatusId.UNFOLLOWED, label: EnumExpandStatus.UNFOLLOWED },
  { value: EnumExpandStatusId.CHANGE_REQUEST, label: EnumExpandStatus.CHANGE_REQUEST },
  { value: EnumExpandStatusId.TO_BE_EVALUATED, label: EnumExpandStatus.TO_BE_EVALUATED },
]

export const newsStatus = [
  { value: EnumCompanyNewsStatusId.FOLLOWING, label: EnumCompanyNewsStatus.FOLLOWING },
  { value: EnumCompanyNewsStatusId.UNFOLLOWED, label: EnumCompanyNewsStatus.UNFOLLOWED },
]

export const leadInvestor = [
  { value: EnumBoolean.TRUE, label: EnumInvestor.LEAD },
  { value: EnumBoolean.FALSE, label: EnumInvestor.SUPPORT },
]

export const fctStatusOptions = [
  { value: EnumExpandStatusId.FOLLOWING, label: EnumExpandStatus.FOLLOWING },
  { value: EnumExpandStatusId.DUPLICATED, label: EnumExpandStatus.DUPLICATED },
]

export const roundType1 = [
  { label: 'Debt Financing', value: 'Debt Financing' },
  { label: 'Equity Financing', value: 'Equity Financing' },
  { label: 'ICO', value: 'ICO' },
  { label: 'Post IPO', value: 'Post IPO' },
  { label: 'Other', value: 'Other' },
]

export const roundType2: Record<string, FormOption[]> = {
  'Debt Financing': [
    { label: 'Debt', value: 'Debt' },
    { label: 'Loan', value: 'Loan' },
    { label: 'Revenue Finance', value: 'Revenue Finance' },
  ],
  'Equity Financing': [
    { label: 'Competition/Grant', value: 'Competition/Grant' },
    { label: 'Corporate', value: 'Corporate' },
    { label: 'Private/Growth Equity', value: 'Private/Growth Equity' },
    { label: 'Seed/Angel', value: 'Seed/Angel' },
    { label: 'Series A', value: 'Series A' },
    { label: 'Series B', value: 'Series B' },
    { label: 'Series C', value: 'Series C' },
    { label: 'Series D', value: 'Series D' },
    { label: 'Series E+', value: 'Series E+' },
    { label: 'Unattributed', value: 'Unattributed' },
  ],
  ICO: [{ label: 'ICO', value: 'ICO' }],
  'Post IPO': [
    { label: 'Option/Warrant', value: 'Option/Warrant' },
    { label: 'Post IPO Convertible', value: 'Post IPO Convertible' },
    { label: 'Post IPO Debt', value: 'Post IPO Debt' },
    { label: 'Post IPO Equity', value: 'Post IPO Equity' },
    { label: 'Post IPO Secondary', value: 'Post IPO Secondary' },
    { label: 'Shelf Registration', value: 'Shelf Registration' },
  ],
  Other: [
    { label: 'Buyback', value: 'Buyback' },
    { label: 'Competition/Grant', value: 'Competition/Grant' },
    { label: 'Corporate-Venture Partnership', value: 'Corporate-Venture Partnership' },
    { label: 'Crowdfunding', value: 'Crowdfunding' },
    { label: 'Dead', value: 'Dead' },
    { label: 'Grant', value: 'Grant' },
    { label: 'Incubator/Accelerator', value: 'Incubator/Accelerator' },
    { label: 'In-kind/Services', value: 'In-kind/Services' },
    { label: 'Media for equity', value: 'Media for equity' },
    { label: 'Portfolio Merger', value: 'Portfolio Merger' },
    { label: 'Portfolio NewCo', value: 'Portfolio NewCo' },
    { label: 'Recap', value: 'Recap' },
    { label: 'Secondary Market', value: 'Secondary Market' },
    { label: 'Shareholder Liquidity', value: 'Shareholder Liquidity' },
    { label: 'Spinoff/Spinout', value: 'Spinoff/Spinout' },
    { label: 'PIPE', value: 'PIPE' },
  ],
}

export const investor = [
  { label: 'Academic/Grant', value: 'Academic/Grant' },
  { label: 'Accelerator', value: 'Accelerator' },
  { label: 'Angel', value: 'Angel' },
  { label: 'Bank', value: 'Bank' },
  { label: 'Corporate', value: 'Corporate' },
  { label: 'Crowdfunding', value: 'Crowdfunding' },
  { label: 'Exchange', value: 'Exchange' },
  { label: 'Government/Nonprofit', value: 'Government/Nonprofit' },
  { label: 'Incubator', value: 'Incubator' },
  { label: 'Information Service Provider', value: 'Information Service Provider' },
  { label: 'Insurance', value: 'Insurance' },
  { label: 'Investment Management', value: 'Investment Management' },
  { label: 'Other', value: 'Other' },
  { label: 'Payments Infrastructure', value: 'Payments Infrastructure' },
  { label: 'PE', value: 'PE' },
  { label: 'Start-Up', value: 'Start-Up' },
  { label: 'Tech', value: 'Tech' },
  { label: 'VC', value: 'VC' },
]

export const ftesRange = [
  { label: '1-10', value: '1-10' },
  { label: '11-50', value: '11-50' },
  { label: '51-100', value: '51-100' },
  { label: '101-250', value: '101-250' },
  { label: '251-500', value: '251-500' },
  { label: '501-1000', value: '501-1000' },
  { label: '1001-5000', value: '1001-5000' },
  { label: '5001-10000', value: '5001-10000' },
  { label: '10001+', value: '10001+' },
]

export const ftesRangeWithBlankOptions = [{ label: 'Blank', value: 'blank' }, ...ftesRange]
