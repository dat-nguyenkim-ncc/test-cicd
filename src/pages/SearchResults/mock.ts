import { companyDetails, companySources, MockMonzoDetails } from '../../__mock__'

export const mock = {
  internal: [
    {
      companyDetails: MockMonzoDetails,
      source: companySources,
    },
    {
      companyDetails: companyDetails(),
      source: 'Source: XPTO',
    },
  ],
  external: [
    {
      companyDetails: companyDetails(),
      source: 'Source: XPTO',
    },
    {
      companyDetails: companyDetails(),
      source: 'Source: XPTO',
    },
    {
      companyDetails: companyDetails(),
      source: 'Source: XPTO',
    },
  ],
}
