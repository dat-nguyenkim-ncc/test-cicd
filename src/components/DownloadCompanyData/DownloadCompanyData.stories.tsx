import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import DownloadCompanyData, { DownloadCompanyDataProps } from './DownloadCompanyData'
import { Box } from 'theme-ui'
import { categoryOptions, mappingOptions } from './mock'

export default {
  title: 'Widgets/DownloadCompanyData',
  component: DownloadCompanyData,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<DownloadCompanyDataProps> = args => (
  <Box sx={{ p: 7, backgroundColor: 'gray03', width: '100%' }}>
    <DownloadCompanyData {...args} />
  </Box>
)

export const Default = Template.bind({})
Default.args = {
  categoryOptions,
  mappingOptions,
  onPressDownload: () => {},
  lastUpdated: new Date().toUTCString(),
} as DownloadCompanyDataProps
