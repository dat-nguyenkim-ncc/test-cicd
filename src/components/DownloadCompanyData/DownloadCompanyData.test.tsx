import * as React from 'react'
import { shallow } from 'enzyme'
import DownloadCompanyData from './DownloadCompanyData'
import { categoryOptions, mappingOptions } from './mock'

describe('DownloadCompanyData', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <DownloadCompanyData
        lastUpdated={new Date().toString()}
        onPressDownload={() => {}}
        categoryOptions={categoryOptions}
        mappingOptions={mappingOptions}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
