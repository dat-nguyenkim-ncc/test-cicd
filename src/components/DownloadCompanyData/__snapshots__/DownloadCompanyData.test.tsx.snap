// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadCompanyData matches snapshot 1`] = `ShallowWrapper {}`;
